src/utils/stark/starkCurve.ts

```diff
@@ -1,23 +1,271 @@
+// DANGER: DO NOT MODIFY GRIND KEY / KEY GENERATION LOGIC.
+//////////////////////////////////////////////////////////////////////////////////////////////////
+/*
+This grindKey function is the default logic to be used by all our SDKs including the imx-sdk-js. All the accounts 
+that were created using imx-sdk-js are to be supported by the legacy.grindKey function.
+
+This function receives a key seed and produces an appropriate StarkEx key from a uniform
+distribution. 
+
+Although it is possible to define a StarkEx key as a residue between the StarkEx EC order and a
+random 256bit digest value, the result would be a biased key. In order to prevent this bias, we
+deterministically search (by applying more hashes, AKA grinding) for a value lower than the largest
+256bit multiple of StarkEx EC order.
+
+https://github.com/starkware-libs/starkware-crypto-utils/blob/dev/src/js/key_derivation.js#L119
+
+The accounts that has been created with versions released between 1.0.0-beta.3 and 2.0.0 are fine as it uses this method. 
+Although this logic has been in place in core-sdk since the following commit, it was momentarily changed in version 2.0.1 to use gringKeyV201 method.
+https://github.com/immutable/imx-core-sdk/commit/3d9e35b4598784589bd7f121f26e105493196716
+*/
+export function grindKey(keySeed: BN) {
+  const maxAllowedVal = MAX_ALLOWED_VAL();
+  // The key passed to hashKeyWithIndex must have a length of 64 characters
+  // to ensure that the correct number of leading zeroes are used as input
+  // to the hashing loop
+  let key = hashKeyWithIndex(keySeed.toString('hex', 64), 0);
+
+  // Make sure the produced key is devided by the Stark EC order, and falls within the range
+  // [0, maxAllowedVal).
+  for (let i = 1; key.gte(maxAllowedVal); i++) {
+    key = hashKeyWithIndex(key.toString('hex'), i);
+  }
+  return key.umod(starkEcOrder).toString('hex');
+}
+
+// Check if the hash value of the the given PrivateKey falls above the starkEcOrder limit.
+// This function is only serving the context of DX-2184, used to determine if we need to validate the generated key
+// against the one recorded in IMX servers.
+export function checkIfHashedKeyIsAboveLimit(keySeed: BN) {
+  const maxAllowedVal = MAX_ALLOWED_VAL();
+  // The key passed to hashKeyWithIndex must have a length of 64 characters
+  // to ensure that the correct number of leading zeroes are used as input
+  // to the hashing loop
+  const key = hashKeyWithIndex(keySeed.toString('hex', 64), 0);
+  return key.gte(maxAllowedVal);
+}
+
+// Gets the account (stark public key) value of the requested user (ethAddress) for Production environment only.
+export async function getStarkPublicKeyFromImx(
+  ethAddress: string,
+): Promise<{ starkPublicKey: string; accountNotFound: boolean } | undefined> {
+  try {
+    const imxClient = new ImmutableX(Config.PRODUCTION);
+    // Check if the generated stark public key matches with the existing account value for that user.
```

Comment:
This comment seems misplaced. The line below only queries and returns if it's been registered, but it doesn't match the generated/local key in this function.

Reply:
Yup, you are right. Will fix it.